fs/osfs.py

```diff
@@ -393,6 +398,43 @@ def _scandir(self, path, namespaces=None):
 
                 yield Info(info)
 
+    def _scandir_py(self, path, namespaces=None):
+        self.check()
+        namespaces = namespaces or ()
+        _path = self.validatepath(path)
+        sys_path = self._to_sys_path(_path)
+        with convert_os_errors('scandir', path, directory=True):
+            for entry_name in os.listdir(sys_path):
+                entry_path = os.path.join(sys_path, entry_name)
+                info = {
+                    "basic": {
+                        "name": entry_name,
+                        "is_dir": os.path.isdir(entry_path),
+                    }
+                }
+                if 'details' in namespaces:
+                    stat_result = os.stat(entry_path)
```

Comment:
Could you cache this `os.stat` call?  It could potentially be called several time. And perhaps you can use the `stat` structure rather than `isdir` on L412, because that will do a stat under the hood.